From 8de1fc964b0ff7f88e42179ec2c30c45642a38da Mon Sep 17 00:00:00 2001
From: Olliver Schinagl <oliver@schinagl.nl>
Date: Wed, 14 Aug 2024 08:58:16 +0200
Subject: [PATCH 16/17] sdk: Disable child aging leave for unsupported devices

Some devices do not support child aging, and would be kicked off the
network because of it. For these devices, disable child leaving.

Signed-off-by: Olliver Schinagl <oliver@schinagl.nl>
---
 source/ti/zstack/startup/zstackstartup.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/source/ti/zstack/startup/zstackstartup.c b/source/ti/zstack/startup/zstackstartup.c
index 88acb5bf5..3b601d959 100644
--- a/source/ti/zstack/startup/zstackstartup.c
+++ b/source/ti/zstack/startup/zstackstartup.c
@@ -701,6 +701,14 @@ static void stackInit(void)
     //Initialize default poll rates
     nwk_InitializeDefaultPollRates();
 
+    // Use custom function for child aging leave requests
+    pNwkNotMyChildSendLeave = &NwkNotMyChildSendLeaveCustom;
+
+    // Disable child aging leave for Xiaomi/Aqara extAddr range to prevent them from being kicekd out of the network.
+    // They do not support child aging.
+    uint8_t extAddrXiaomi [] = {0x00, 0x00, 0x00, 0x00, 0x00, 0x8d, 0x15, 0x00};
+    NwkDisableChildAgingLeaveAdd(extAddrXiaomi, 3);
+
     /* Initialize MAC buffer */
     macLowLevelBufferInit();
 
-- 
2.46.0

